src/state/apis/swapper/getUsdRatesApi.ts

```diff
@@ -0,0 +1,84 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import type { GetTradeQuoteInput } from '@shapeshiftoss/swapper'
+import type { GetTradeQuoteInputArgs } from 'components/Trade/hooks/useSwapper/getTradeQuoteArgs'
+import { getTradeQuoteArgs } from 'components/Trade/hooks/useSwapper/getTradeQuoteArgs'
+import { getBestSwapperApi } from 'state/apis/swapper/getBestSwapperApi'
+import { getUsdRateApi } from 'state/apis/swapper/getUsdRateApi'
+import { swapperApi } from 'state/apis/swapper/swapperApi'
+import type { State } from 'state/apis/types'
+
+export type GetUsdRatesArgs = {
+  feeAssetId: AssetId
+} & (
+  | { tradeQuoteInputArgs?: never; tradeQuoteArgs: GetTradeQuoteInput }
+  | { tradeQuoteInputArgs: GetTradeQuoteInputArgs; tradeQuoteArgs?: never }
+)
+
+type GetUsdRatesReturn = {
+  buyAssetUsdRate: string
+  sellAssetUsdRate: string
+  feeAssetUsdRate: string
+}
+
+const getBestSwapperType = getBestSwapperApi.endpoints.getBestSwapperType
+const getUsdRate = getUsdRateApi.endpoints.getUsdRate
+
+export const getUsdRatesApi = swapperApi.injectEndpoints({
```

Comment:
this is neat!

Reply:
Brings us closer to RTK query best practice of "a single "API slice" per application".